lib/android/app/src/main/java/com/reactnativenavigation/viewcontrollers/modal/Modal.java

```diff
@@ -19,24 +23,44 @@
     private ModalListener modalListener;
     @Nullable private Promise dismissPromise;
 
+    private ModalAnimator animator;
+
     public Modal(final ViewController viewController, ModalListener modalListener) {
         this.viewController = viewController;
         dialog = new Dialog(viewController.getActivity(), R.style.Modal);
         this.modalListener = modalListener;
         dialog.setOnKeyListener(this);
         dialog.setOnDismissListener(this);
         dialog.setOnShowListener(this);
+        animator = new ModalAnimator(viewController.getActivity());
+        animator.setOptions(viewController.options.animationsOptions);
     }
 
     public void show() {
         preMeasureView();
-        dialog.setContentView(viewController.getView());
+        final View contentView = viewController.getView();
         dialog.show();
+        animator.animateShow(contentView, new AnimatorListenerAdapter() {
+            @Override
+            public void onAnimationStart(Animator animation) {
+                dialog.setContentView(contentView);
+            }
+        });
+    }
+
+    @RestrictTo(RestrictTo.Scope.TESTS)
```

Comment:
This method needs to be remove. We can't add methods or any logic for tests.
The only acceptable use for `RestrictTo(RestrictTo.Scope.TESTS)` is for getters/setters. This way we avoid using tools like [Power Mock](https://github.com/powermock/powermock).